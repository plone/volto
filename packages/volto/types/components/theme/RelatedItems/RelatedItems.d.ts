export default RelatedItems;
/**
 * Related Items component.
 * @function RelatedItems
 * @param {array} relatedItems Array of related items.
 * @returns {JSX.Element} Markup of the component.
 */
declare function RelatedItems({ content }: any[]): JSX.Element;
declare namespace RelatedItems {
    namespace propTypes {
        let content: any;
    }
}
