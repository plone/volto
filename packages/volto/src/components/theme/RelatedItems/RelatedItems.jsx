/**
 * RelatedItems component.
 * @module components/theme/RelatedItems/RelatedItems
 */

import UniversalLink from '@plone/volto/components/manage/UniversalLink/UniversalLink';
import { defineMessages, useIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import config from '@plone/volto/registry';

const messages = defineMessages({
  relatedItems: {
    id: 'Related Items',
    defaultMessage: 'Related Items',
  },
});

/**
 * Related Items component.
 * @function RelatedItems
 * @param {array} relatedItems Array of related items.
 * @returns {JSX.Element} Markup of the component.
 */
const RelatedItems = ({ content }) => {
  const intl = useIntl();
  const relatedItems = content?.relatedItems;
  if (
    !config.settings.showRelatedItems ||
    !relatedItems ||
    relatedItems.length === 0
  ) {
    return null;
  }

  return (
    <Container className="related-items">
      <h2>{intl.formatMessage(messages.relatedItems)}</h2>
      <ul>
        {relatedItems.map((relatedItem) =>
          relatedItem ? (
            <li key={relatedItem['@id']}>
              <UniversalLink href={relatedItem['@id']}>
                {relatedItem.title}
              </UniversalLink>
            </li>
          ) : null,
        )}
      </ul>
    </Container>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
RelatedItems.propTypes = {
  content: PropTypes.shape({
    relatedItems: PropTypes.arrayOf(
      PropTypes.shape({
        '@id': PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }),
    ),
  }),
};

export default RelatedItems;
