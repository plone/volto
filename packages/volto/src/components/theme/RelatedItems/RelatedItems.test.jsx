import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import RelatedItems from './RelatedItems';

const mockStore = configureStore();
let store;

describe('Related Items', () => {
  beforeEach(() => {
    store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
  });

  it('renders without related items', () => {
    const content = {};
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with related items', () => {
    const content = {
      relatedItems: [
        {
          '@id': '/test-1',
          title: 'Title 1',
          description: 'Description 1',
        },
        {
          '@id': '/test-2',
          title: 'Title 2',
          description: 'Description 2',
        },
      ],
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with related items has null', () => {
    const content = {
      relatedItems: [
        {
          '@id': '/test-1',
          title: 'Title 1',
          description: 'Description 1',
        },
        null,
      ],
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
