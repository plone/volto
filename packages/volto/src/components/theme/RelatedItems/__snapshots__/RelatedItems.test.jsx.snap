// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Items renders with related items 1`] = `
<div
  className="ui container related-items"
>
  <h2>
    Related Items
  </h2>
  <ul>
    <li>
      <a
        className={null}
        href="/test-1"
        onClick={[Function]}
        target={null}
        title={null}
      >
        Title 1
      </a>
    </li>
    <li>
      <a
        className={null}
        href="/test-2"
        onClick={[Function]}
        target={null}
        title={null}
      >
        Title 2
      </a>
    </li>
  </ul>
</div>
`;

exports[`Related Items renders with related items has null 1`] = `
<div
  className="ui container related-items"
>
  <h2>
    Related Items
  </h2>
  <ul>
    <li>
      <a
        className={null}
        href="/test-1"
        onClick={[Function]}
        target={null}
        title={null}
      >
        Title 1
      </a>
    </li>
  </ul>
</div>
`;

exports[`Related Items renders without related items 1`] = `null`;
